bin/metaBackend.py

```diff
@@ -21,47 +21,506 @@
+        @param level         level (e.g., L1, L2, L3)
+        @param dataRel       data release
+        @param owner         owner of the database
+        @param accessibility accessibility of the database (pending/public/private)
+        @param projectName   name of the project the db is associated with
+        @param dbMysqlAuthF  mysql auth file for the db that we are adding
+
+        The function connects to two database servers:
+        a) one that has the database that is being loaded
+        b) one that has the metaserv database
+        If they are both on the same server, the connection is reused.
+
+        The course of action:
+        * connect to the server that has database that is being loaded
+        * parse the ascii schema file
+        * fetch schema information from the information_schema
+        * do the matching, add info fetched from information_schema to the
+          in memory structure produced by parsing ascii schema file
+        * fetch schema description and version (which is kept as data inside
+          a special table in the database that is being loaded). Ignore if it
+          does not exist.
+        * Capture information from mysql auth file about connection information
+        * connect to the metaserv database
+        * validate owner, project (these must be loaded into metaserv prior to
+        calling this function)
+        * load all the information into metaserv in various tables (Repo,
+          DDT_Table, DDT_Column)
+
+        It raises following MetaBEXceptions:
+        * DB_DOES_NOT_EXISTS if database dbName does not exist
+        * NOT_MATCHING if the database schema and ascii schema don't match
+        * TB_NOT_IN_DB if the table is described in ascii schema, but it is missing
+                       in the database
+        * COL_NOT_IN_TB if the column is described in ascii schema, but it is
+                        missing in the database
+        * COL_NOT_IN_FL if the column is in the database schema, but not in ascii
+                        schema
+        * Db object can throw various DbException and MySQL exceptions
+        """
+
+        # Connect to the server that has database that is being added
+        db = Db(read_default_file=dbMysqlAuthF)
+        if not db.dbExists(dbName):
+            self._log.error("Db '%s' not found.", dbName)
+            raise MetaBException(MetaBException.DB_DOES_NOT_EXIST, dbName)
+
+        # Parse the ascii schema file
+        x = SchemaToMeta(schemaFile)
+        theTable = x.parse()
+
+        # Fetch the schema information from the database
+        ret = db.execCommandN(
+            "SELECT table_name, column_name, ordinal_position "
+            "FROM information_schema.COLUMNS WHERE "
+            "TABLE_SCHEMA = %s ORDER BY table_name", (dbName,))
+
+        # Count the number of columns in the ascii file
+        nColumns = 0
+
+        for t in theTable:
+            nColumns += len(theTable[t]["columns"])
```

Comment:
`nColumns = sum(len(t["columns"]) for t in theTable.values())` may be more readable and efficient